<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
<meta name="viewport" content="width=device-width">
<meta name="theme-color" content="#222"><meta name="generator" content="Hexo 7.3.0">

  <link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon-next.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/images/favicon-32x32-next.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/images/favicon-16x16-next.png">
  <link rel="mask-icon" href="/images/logo.svg" color="#222">

<link rel="stylesheet" href="/css/main.css">



<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.7.2/css/all.min.css" integrity="sha256-dABdfBfUoC8vJUBOwGVdm8L9qlMWaHTIfXt+7GnZCIo=" crossorigin="anonymous">
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/animate.css/3.1.1/animate.min.css" integrity="sha256-PR7ttpcvz8qrF57fur/yAx1qXMFJeJFiA6pSzWi0OIE=" crossorigin="anonymous">

<script class="next-config" data-name="main" type="application/json">{"hostname":"lingjunm.github.io","root":"/","images":"/images","scheme":"Pisces","darkmode":false,"version":"8.23.2","exturl":false,"sidebar":{"position":"left","width_expanded":320,"width_dual_column":240,"display":"post","padding":18,"offset":12},"hljswrap":true,"codeblock":{"theme":{"light":"default","dark":"stackoverflow-dark"},"prism":{"light":"prism","dark":"prism-dark"},"copy_button":{"enable":false,"style":null},"fold":{"enable":false,"height":500},"language":false},"bookmark":{"enable":false,"color":"#222","save":"auto"},"mediumzoom":false,"lazyload":false,"pangu":false,"comments":{"style":"tabs","active":null,"storage":true,"lazyload":false,"nav":null},"stickytabs":false,"motion":{"enable":true,"async":false,"duration":200,"transition":{"menu_item":"fadeInDown","post_block":"fadeIn","post_header":"fadeInDown","post_body":"fadeInDown","coll_header":"fadeInLeft","sidebar":"fadeInUp"}},"i18n":{"placeholder":"Searching...","empty":"We didn't find any results for the search: ${query}","hits_time":"${hits} results found in ${time} ms","hits":"${hits} results found"}}</script><script src="/js/config.js" defer></script>

    <meta name="description" content="分析CRISPR文库的病毒感染效率">
<meta property="og:type" content="article">
<meta property="og:title" content="CRISPR文库测序数据分析">
<meta property="og:url" content="https://lingjunm.github.io/2025/07/16/CRISPR%E6%96%87%E5%BA%93%E6%B5%8B%E5%BA%8F%E6%95%B0%E6%8D%AE%E5%88%86%E6%9E%90/index.html">
<meta property="og:site_name" content="读研日记">
<meta property="og:description" content="分析CRISPR文库的病毒感染效率">
<meta property="og:locale" content="en_US">
<meta property="article:published_time" content="2025-07-16T02:11:37.000Z">
<meta property="article:modified_time" content="2025-08-20T01:34:49.307Z">
<meta property="article:author" content="Meng">
<meta property="article:tag" content="CRISPR">
<meta property="article:tag" content="NGS">
<meta property="article:tag" content="Bowtie2">
<meta property="article:tag" content="samtools">
<meta property="article:tag" content="fastqc">
<meta property="article:tag" content="cutadapt">
<meta property="article:tag" content="ggplot2">
<meta property="article:tag" content="FASTA">
<meta property="article:tag" content="pipeline">
<meta property="article:tag" content="data ana">
<meta name="twitter:card" content="summary">


<link rel="canonical" href="https://lingjunm.github.io/2025/07/16/CRISPR%E6%96%87%E5%BA%93%E6%B5%8B%E5%BA%8F%E6%95%B0%E6%8D%AE%E5%88%86%E6%9E%90/">


<script class="next-config" data-name="page" type="application/json">{"sidebar":"","isHome":false,"isPost":true,"lang":"en","comments":true,"permalink":"https://lingjunm.github.io/2025/07/16/CRISPR%E6%96%87%E5%BA%93%E6%B5%8B%E5%BA%8F%E6%95%B0%E6%8D%AE%E5%88%86%E6%9E%90/","path":"2025/07/16/CRISPR文库测序数据分析/","title":"CRISPR文库测序数据分析"}</script>

<script class="next-config" data-name="calendar" type="application/json">""</script>
<title>CRISPR文库测序数据分析 | 读研日记</title>
  








  
  <script src="https://cdnjs.cloudflare.com/ajax/libs/animejs/3.2.1/anime.min.js" integrity="sha256-XL2inqUJaslATFnHdJOi9GfQ60on8Wx1C2H8DYiN1xY=" crossorigin="anonymous" defer></script>
<script src="/js/utils.js" defer></script><script src="/js/motion.js" defer></script><script src="/js/sidebar.js" defer></script><script src="/js/next-boot.js" defer></script>

  






  





  <noscript>
    <link rel="stylesheet" href="/css/noscript.css">
  </noscript>
</head>

<body itemscope itemtype="http://schema.org/WebPage" class="use-motion">
  <div class="headband"></div>

  <main class="main">
    <div class="column">
      <header class="header" itemscope itemtype="http://schema.org/WPHeader"><div class="site-brand-container">
  <div class="site-nav-toggle">
    <div class="toggle" aria-label="Toggle navigation bar" role="button">
        <span class="toggle-line"></span>
        <span class="toggle-line"></span>
        <span class="toggle-line"></span>
    </div>
  </div>

  <div class="site-meta">

    <a href="/" class="brand" rel="start">
      <i class="logo-line"></i>
      <p class="site-title">读研日记</p>
      <i class="logo-line"></i>
    </a>
  </div>

  <div class="site-nav-right">
    <div class="toggle popup-trigger" aria-label="Search" role="button">
    </div>
  </div>
</div>



<nav class="site-nav">
  <ul class="main-menu menu"><li class="menu-item menu-item-home"><a href="/" rel="section"><i class="fa fa-home fa-fw"></i>Home</a></li><li class="menu-item menu-item-about"><a href="/about/" rel="section"><i class="fa fa-user fa-fw"></i>About</a></li><li class="menu-item menu-item-tags"><a href="/tags/" rel="section"><i class="fa fa-tags fa-fw"></i>Tags</a></li><li class="menu-item menu-item-categories"><a href="/categories/" rel="section"><i class="fa fa-th fa-fw"></i>Categories</a></li><li class="menu-item menu-item-archives"><a href="/archives/" rel="section"><i class="fa fa-archive fa-fw"></i>Archives</a></li>
  </ul>
</nav>




</header>
        
  
  <aside class="sidebar">

    <div class="sidebar-inner sidebar-nav-active sidebar-toc-active">
      <ul class="sidebar-nav">
        <li class="sidebar-nav-toc">
          Table of Contents
        </li>
        <li class="sidebar-nav-overview">
          Overview
        </li>
      </ul>

      <div class="sidebar-panel-container">
        <!--noindex-->
        <div class="post-toc-wrap sidebar-panel">
            <div class="post-toc animated"><ol class="nav"><li class="nav-item nav-level-1"><a class="nav-link" href="#%E5%88%B6%E4%BD%9C%E5%8F%82%E8%80%83%E5%BA%8F%E5%88%97%E6%96%87%E4%BB%B6"><span class="nav-number">1.</span> <span class="nav-text">制作参考序列文件</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#%E5%88%86%E5%89%B2sgRNA"><span class="nav-number">2.</span> <span class="nav-text">分割sgRNA</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#%E6%AF%94%E5%AF%B9%E8%BF%87%E6%BB%A4"><span class="nav-number">3.</span> <span class="nav-text">比对过滤</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#%E7%BB%9F%E8%AE%A1"><span class="nav-number">4.</span> <span class="nav-text">统计</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#R%E4%BD%9C%E5%9B%BE"><span class="nav-number">5.</span> <span class="nav-text">R作图</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#2"><span class="nav-number">6.</span> <span class="nav-text">2</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#%E5%90%88%E5%B9%B6kmer"><span class="nav-number">7.</span> <span class="nav-text">合并kmer</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#%E6%9D%8E%E8%80%81%E5%B8%88-%E8%A7%A3%E7%A0%81%E7%94%9F%E7%89%A9%E6%96%87%E5%BA%93%E6%AF%94%E8%BE%83"><span class="nav-number">8.</span> <span class="nav-text">李老师 解码生物文库比较</span></a></li></ol></div>
        </div>
        <!--/noindex-->

        <div class="site-overview-wrap sidebar-panel">
          <div class="site-author animated" itemprop="author" itemscope itemtype="http://schema.org/Person">
  <p class="site-author-name" itemprop="name">Meng</p>
  <div class="site-description" itemprop="description">生物学拾遗</div>
</div>
<div class="site-state-wrap animated">
  <nav class="site-state">
      <div class="site-state-item site-state-posts">
        <a href="/archives/">
          <span class="site-state-item-count">7</span>
          <span class="site-state-item-name">posts</span>
        </a>
      </div>
      <div class="site-state-item site-state-tags">
          <a href="/tags/">
        <span class="site-state-item-count">17</span>
        <span class="site-state-item-name">tags</span></a>
      </div>
  </nav>
</div>

        </div>
      </div>
    </div>

    
  </aside>


    </div>

    <div class="main-inner post posts-expand">


  


<div class="post-block">
  
  

  <article itemscope itemtype="http://schema.org/Article" class="post-content" lang="en">
    <link itemprop="mainEntityOfPage" href="https://lingjunm.github.io/2025/07/16/CRISPR%E6%96%87%E5%BA%93%E6%B5%8B%E5%BA%8F%E6%95%B0%E6%8D%AE%E5%88%86%E6%9E%90/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.gif">
      <meta itemprop="name" content="Meng">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="读研日记">
      <meta itemprop="description" content="生物学拾遗">
    </span>

    <span hidden itemprop="post" itemscope itemtype="http://schema.org/CreativeWork">
      <meta itemprop="name" content="CRISPR文库测序数据分析 | 读研日记">
      <meta itemprop="description" content="">
    </span>
      <header class="post-header">
        <h1 class="post-title" itemprop="name headline">
          CRISPR文库测序数据分析
        </h1>

        <div class="post-meta-container">
          <div class="post-meta">
    <span class="post-meta-item">
      <span class="post-meta-item-icon">
        <i class="far fa-calendar"></i>
      </span>
      <span class="post-meta-item-text">Posted on</span>

      <time title="Created: 2025-07-16 10:11:37" itemprop="dateCreated datePublished" datetime="2025-07-16T10:11:37+08:00">2025-07-16</time>
    </span>
    <span class="post-meta-item">
      <span class="post-meta-item-icon">
        <i class="far fa-calendar-check"></i>
      </span>
      <span class="post-meta-item-text">Edited on</span>
      <time title="Modified: 2025-08-20 09:34:49" itemprop="dateModified" datetime="2025-08-20T09:34:49+08:00">2025-08-20</time>
    </span>

  
</div>

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody"><p>分析CRISPR文库的病毒感染效率 </p>
<span id="more"></span>

<h1 id="制作参考序列文件"><a href="#制作参考序列文件" class="headerlink" title="制作参考序列文件"></a>制作参考序列文件</h1><p>通常测序比对时，参考基因组是鼠&#x2F;人等生物的基因组。<br>sgRNA测序只需要检测是否转入对应的20 bp sgRNA序列即可，因此参考基因组为设计好的sgRNA序列。<br>含有sgRNA序列的 excel 示例</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">POS_RPS19_0006,TATCTTGTGGAAAGGACGAAACACCGTGTCAGTTTGCGGCCGCTGTGTTTAAGAGCTATGCTGGAAACAGCATAG</span><br></pre></td></tr></table></figure>

<p>参考基因组的文件是FASTA格式，第一行是‘&gt;’加‘染色体名’，后一行是序列。需要手动制作该文件。</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br></pre></td><td class="code"><pre><span class="line">file <span class="operator">=</span> readxl<span class="operator">::</span>read_xlsx<span class="punctuation">(</span><span class="string">&quot;../data/library_oligo.xlsx&quot;</span><span class="punctuation">)</span></span><br><span class="line"></span><br><span class="line">gene_name <span class="operator">=</span> paste0<span class="punctuation">(</span><span class="string">&quot;&gt;&quot;</span><span class="punctuation">,</span>file<span class="operator">$</span>sgName<span class="punctuation">)</span></span><br><span class="line">seq <span class="operator">=</span> file<span class="operator">$</span>`oligo for synthesis`</span><br><span class="line"></span><br><span class="line">fasta_file <span class="operator">&lt;-</span> file<span class="punctuation">(</span><span class="string">&quot;sgRNA.fasta&quot;</span><span class="punctuation">,</span> <span class="string">&quot;w&quot;</span><span class="punctuation">)</span></span><br><span class="line"></span><br><span class="line"><span class="keyword">for</span> <span class="punctuation">(</span>i <span class="keyword">in</span> <span class="number">1</span><span class="operator">:</span>nrow<span class="punctuation">(</span>file<span class="punctuation">)</span><span class="punctuation">)</span> <span class="punctuation">&#123;</span></span><br><span class="line">  <span class="comment"># 提取名和序列</span></span><br><span class="line">  protein_name <span class="operator">&lt;-</span> gene_name<span class="punctuation">[</span>i<span class="punctuation">]</span></span><br><span class="line">  sequence     <span class="operator">&lt;-</span> seq<span class="punctuation">[</span>i<span class="punctuation">]</span></span><br><span class="line">  </span><br><span class="line">  <span class="comment"># 写入 FASTA 标题行和序列行</span></span><br><span class="line">  writeLines<span class="punctuation">(</span>protein_name<span class="punctuation">,</span> fasta_file<span class="punctuation">)</span></span><br><span class="line">  writeLines<span class="punctuation">(</span>sequence<span class="punctuation">,</span> fasta_file<span class="punctuation">)</span></span><br><span class="line"><span class="punctuation">&#125;</span></span><br><span class="line"></span><br><span class="line">close<span class="punctuation">(</span>fasta_file<span class="punctuation">)</span></span><br></pre></td></tr></table></figure>

<p>比对软件还需要参考文件的索引，制作索引。</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">bowtie2-build --threads 6 -f sgRNA.fasta sgrna </span><br></pre></td></tr></table></figure>

<h1 id="分割sgRNA"><a href="#分割sgRNA" class="headerlink" title="分割sgRNA"></a>分割sgRNA</h1><p>PCR之后的sgRNA文库，长度为466bp   只需关注20bp的sgRNA部分，位于前150bp内<br>检查双端测序的结果，只需使用一端测序的文件，另一端测序的结果不包含sgRNA部分。<br>使用cutadapt软件仅保留20bp的sgRNA部分</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">cutadapt -u 83 -u -47 -o trim/i20.fq data/E250704001_L01_Lib-2P-Index1_1.fq.gz</span><br><span class="line"><span class="meta prompt_"></span></span><br><span class="line"><span class="meta prompt_"># </span><span class="language-bash">检查质量</span></span><br><span class="line">fastqc -t 10 -o qc/before  data/*sample1_1.fq.gz &gt;qc/before/fastqc.beforeTrim.log 2&gt;&amp;1</span><br><span class="line">fastqc -t 10 -o qc/after/  trim/i20.fq &gt;qc/after/fastqc.afterTrim.log 2&gt;&amp;1</span><br></pre></td></tr></table></figure>

<h1 id="比对过滤"><a href="#比对过滤" class="headerlink" title="比对过滤"></a>比对过滤</h1><figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">bowtie2 -x ~/ref/sgRNA/sgrna -U trim/i20.fq -p 10 --very-sensitive|samtools sort -o bam/i20.sort.bam -O BAM -@ 8</span><br><span class="line">samtools view -q 10 bam/i20.sort.bam -o bam/i20.q10.sort.bam</span><br></pre></td></tr></table></figure>

<h1 id="统计"><a href="#统计" class="headerlink" title="统计"></a>统计</h1><p>samtools的idxstats命令可以统计bam文件中比对到每条‘染色体’的序列数</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">samtools index bam/i20.q10.sort.bam</span><br><span class="line">samtools idxstats  bam/i20.q10.sort.bam | cut -f 1,3 &gt; sam.idx.count.txt</span><br></pre></td></tr></table></figure>

<h1 id="R作图"><a href="#R作图" class="headerlink" title="R作图"></a>R作图</h1><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 读取文件  </span></span><br><span class="line">counts <span class="operator">=</span> read.table<span class="punctuation">(</span><span class="string">&quot;sam.idx.count.txt&quot;</span><span class="punctuation">)</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># idxstats生成的最后一行无意义  重命名列</span></span><br><span class="line">counts <span class="operator">=</span> counts<span class="punctuation">[</span><span class="operator">-</span>nrow<span class="punctuation">(</span>counts<span class="punctuation">)</span><span class="punctuation">,</span><span class="punctuation">]</span></span><br><span class="line">colnames<span class="punctuation">(</span>counts<span class="punctuation">)</span> <span class="operator">=</span> <span class="built_in">c</span><span class="punctuation">(</span><span class="string">&quot;sgRNAname&quot;</span><span class="punctuation">,</span><span class="string">&quot;count&quot;</span><span class="punctuation">)</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># 统计成功转入基因组的sgRNA</span></span><br><span class="line"><span class="built_in">sum</span><span class="punctuation">(</span>counts<span class="operator">$</span>count<span class="operator">&gt;</span><span class="number">0</span><span class="punctuation">)</span></span><br><span class="line"><span class="built_in">sum</span><span class="punctuation">(</span>counts<span class="operator">$</span>count<span class="operator">&gt;</span><span class="number">10</span><span class="punctuation">)</span></span><br><span class="line"><span class="built_in">sum</span><span class="punctuation">(</span>counts<span class="operator">$</span>count<span class="operator">&gt;</span><span class="number">100</span><span class="punctuation">)</span></span><br><span class="line"><span class="built_in">sum</span><span class="punctuation">(</span>counts<span class="operator">$</span>count<span class="operator">&gt;</span><span class="number">1000</span><span class="punctuation">)</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># lg2 变换</span></span><br><span class="line"><span class="built_in">range</span><span class="punctuation">(</span>counts<span class="operator">$</span>count<span class="punctuation">)</span></span><br><span class="line">counts<span class="operator">$</span>lg2 <span class="operator">=</span> log2<span class="punctuation">(</span>counts<span class="operator">$</span>count <span class="operator">+</span> <span class="number">1</span><span class="punctuation">)</span></span><br><span class="line"><span class="built_in">range</span><span class="punctuation">(</span>counts<span class="operator">$</span>lg2<span class="punctuation">)</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># 制图</span></span><br><span class="line">p <span class="operator">=</span> ggplot<span class="punctuation">(</span>counts<span class="punctuation">,</span>aes<span class="punctuation">(</span>lg2<span class="punctuation">)</span><span class="punctuation">)</span> <span class="operator">+</span></span><br><span class="line">  coord_cartesian<span class="punctuation">(</span>xlim <span class="operator">=</span> <span class="built_in">c</span><span class="punctuation">(</span><span class="operator">-</span><span class="number">1</span><span class="punctuation">,</span> <span class="number">15</span><span class="punctuation">)</span><span class="punctuation">)</span> <span class="operator">+</span></span><br><span class="line">  geom_histogram<span class="punctuation">(</span>binwidth <span class="operator">=</span> <span class="number">0.1</span><span class="punctuation">)</span> <span class="operator">+</span></span><br><span class="line">  xlab<span class="punctuation">(</span><span class="string">&quot;&quot;</span><span class="punctuation">)</span><span class="operator">+</span></span><br><span class="line">  ylab<span class="punctuation">(</span><span class="string">&quot;&quot;</span><span class="punctuation">)</span><span class="operator">+</span></span><br><span class="line">  <span class="comment">#xlab(expression(&quot;Log&quot;[2]*&quot;Counts&quot;)) +</span></span><br><span class="line">  <span class="comment">#ylab(&quot;sgRNA Number&quot;)+</span></span><br><span class="line">  theme_classic<span class="punctuation">(</span><span class="punctuation">)</span><span class="operator">+</span></span><br><span class="line">  theme<span class="punctuation">(</span>axis.text <span class="operator">=</span> element_text<span class="punctuation">(</span>size <span class="operator">=</span> <span class="number">18</span><span class="punctuation">)</span><span class="punctuation">)</span></span><br><span class="line"></span><br><span class="line">p</span><br><span class="line"></span><br><span class="line"><span class="comment"># pdf 方便保存查看</span></span><br><span class="line">pdf<span class="punctuation">(</span><span class="string">&quot;sgRNA_volcano.pdf&quot;</span><span class="punctuation">)</span></span><br><span class="line">p</span><br><span class="line">dev.off<span class="punctuation">(</span><span class="punctuation">)</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># png用于PPT汇报</span></span><br><span class="line">png<span class="punctuation">(</span><span class="string">&quot;sgRNA_volcano.png&quot;</span><span class="punctuation">)</span></span><br><span class="line">p</span><br><span class="line">dev.off<span class="punctuation">(</span><span class="punctuation">)</span></span><br></pre></td></tr></table></figure>








<h1 id="2"><a href="#2" class="headerlink" title="2"></a>2</h1><p>cut -f 3 lisglib.txt &gt; lib.seq<br>cut -f 2 lisglib.txt &gt; lib.name<br>cut -f 1 lisglib.txt|sed -E ‘s&#x2F;::[ACGT]{20}(:|$)&#x2F;\1&#x2F;g’  &gt; lib.sg<br>cut -f 1 lisglib.txt|sed -E ‘s&#x2F;::[ACGT]{20}(:|$)&#x2F;\1&#x2F;g’  &gt; lib.sg</p>
<h1 id="合并kmer"><a href="#合并kmer" class="headerlink" title="合并kmer"></a>合并kmer</h1><p>files&#x3D;list.files(pattern&#x3D;”tsv”)<br>kmers &#x3D;list()<br>meta&#x3D;read.table(“..&#x2F;..&#x2F;..&#x2F;library.csv”,sep&#x3D;”,”,col.names&#x3D;c(“sgRNA”,”seq”,”gene”))<br>for (i in 1:6) {kmers[[i]] &#x3D; read.table(files[i],col.names&#x3D;c(“seq”,”counts”)) }</p>
<p>counts &#x3D; merge(meta,kmers[[1]],by&#x3D;”seq”,all.x&#x3D;T,sort&#x3D;F)<br>colnames(counts)[length(colnames(counts))] &#x3D; files[1]<br>for (i in 2:6) {counts &#x3D; merge(counts,kmers[[i]],by&#x3D;”seq”,all.x&#x3D;T,sort&#x3D;F);colnames(counts)[length(colnames(counts))] &#x3D; files[i] }</p>
<p>write.table(counts[,-1],”counts.txt”,quote&#x3D;F,row.names&#x3D;F,sep&#x3D;”\t”)</p>
<p>chr9:110022655-110022998(.)::npk2::pkscore53_014        chr9:110022655-110022998(.)::npk2::pkscore53    0       0       0       0       0       0</p>
<p>sampleList&#x3D;”..&#x2F;..&#x2F;day14vsday0.srr.list”<br>mapfile -t samples &lt; $sampleList</p>
<p>for sample in ${samples[@]}<br>do<br>grep -wf ..&#x2F;..&#x2F;lib.seq ${sample}.kmercount.tsv  &gt;intersect&#x2F;${sample}.intersect.kmercount.tsv<br>done</p>
<h1 id="李老师-解码生物文库比较"><a href="#李老师-解码生物文库比较" class="headerlink" title="李老师 解码生物文库比较"></a>李老师 解码生物文库比较</h1><p>$ seqkit seq -s fq&#x2F;E250812001_L01_Lib-2P-Index1_1.fq.gz|cut -c 1-3|sort|uniq -c<br>  64359 AAA<br>  50278 AAC<br>  57045 AAG<br>    702 AAN<br>23303931 AAT<br>  23932 ACA<br>  13715 ACC<br>   2624 ACG<br>      6 ACN<br>  24948 ACT<br>   6105 AGA<br>   6258 AGC<br>  17822 AGG<br>      4 AGN<br>  47348 AGT<br>    887 ANN<br>   1545 ATA<br>  13817 ATC<br>  88471 ATG<br>      4 ATN<br>   9639 ATT<br>  22248 CAA<br>   5776 CAC<br>   9238 CAG<br>      9 CAN<br>  24276 CAT<br>  25294 CCA<br>  35007 CCC<br>   4777 CCG<br>     15 CCN<br>  44595 CCT<br>   3078 CGA<br>   2126 CGC<br>   2397 CGG<br>      2 CGN<br>   1982 CGT<br>     49 CNN<br>  23139 CTA<br>  38143 CTC<br>  24985 CTG<br>     15 CTN<br>  36016 CTT<br>  18098 GAA<br>  15798 GAC<br>  27329 GAG<br>     10 GAN<br>  42866 GAT<br>  16125 GCA<br>  27174 GCC<br>  18987 GCG<br>     14 GCN<br>  54432 GCT<br>  20806 GGA<br>  22152 GGC<br>  40696 GGG<br>     29 GGN<br>  57471 GGT<br>     87 GNN<br>  60965 GTA<br>  52550 GTC<br>  55268 GTG<br>     29 GTN<br>  22250 GTT<br>     48 NNA<br>     43 NNC<br>     54 NNG<br>    528 NNN<br>   3954 NNT<br>   1299 TAA<br>    999 TAC<br>    813 TAG<br>      1 TAN<br>  26557 TAT<br>   2671 TCA<br>  14486 TCC<br>   1087 TCG<br>      6 TCN<br>  87882 TCT<br>    876 TGA<br>   1433 TGC<br>  15708 TGG<br>      3 TGN<br>   1501 TGT<br>     10 TNN<br>    979 TTA<br>   5097 TTC<br>   5219 TTG<br>      2 TTN<br>   9864 TTT</p>
<pre><code>

# 统计序列信息

$ seqkit fx2tab  -l  -n ../trim5_cutadapt/MCherryR1_read1.fq.gz |cut -f 2|sort|uniq -c
 527892 19
21484743 20

$ seqkit fx2tab  -l  -n ../trim5_cutadapt/MCherryR2_read1.fq.gz |cut -f 2|sort|uniq -c
 585165 19
21415921 20

$ seqkit fx2tab  -l  -n ../trim5_cutadapt/MCherryR3_read1.fq.gz |cut -f 2|sort|uniq -c
 187257 19
6162445 20

$ seqkit fx2tab  -l  -n ../trim5_cutadapt/DoublePosR1_read1.fq.gz |cut -f 2|sort|uniq -c
 962601 19
38361167 20

$ seqkit fx2tab  -l  -n ../trim5_cutadapt/DoublePosR2_read1.fq.gz |cut -f 2|sort|uniq -c
 475111 19
17865918 20

$ seqkit fx2tab  -l  -n ../trim5_cutadapt/DoublePosR3_read1.fq.gz |cut -f 2|sort|uniq -c
 891715 19
29943076 20

$ seqkit fx2tab  -l  -n ../trim5_cutadapt/DoubleNegR1_read1.fq.gz |cut -f 2|sort|uniq -c
1108012 19
41876707 20

$ seqkit fx2tab  -l  -n ../trim5_cutadapt/DoubleNegR2_read1.fq.gz |cut -f 2|sort|uniq -c
1003046 19
38828358 20

$ seqkit fx2tab  -l  -n ../trim5_cutadapt/DoubleNegR3_read1.fq.gz |cut -f 2|sort|uniq -c
 734230 19
25903398 20


$ sed &#39;s/\t/,/g&#39; DoublePosVSMcherry.gene_summary.txt &gt; DoublePosVSMcherry.gene_summary.csv
$ sed &#39;s/\t/,/g&#39; DoublePosVSMcherry.sgrna_summary.txt &gt; DoublePosVSMcherry.sgrna_summary.csv
$ sed &#39;s/\t/,/g&#39; DoubleNegVSMcherry.gene_summary.txt &gt; DoubleNegVSMcherry.gene_summary.csv
$ sed &#39;s/\t/,/g&#39; DoubleNegVSMcherry.sgrna_summary.txt &gt;  DoubleNegVSMcherry.sgrna_summary.csv



</code></pre>

    </div>

    
    
    

    <footer class="post-footer">
          <div class="post-tags">
              <a href="/tags/CRISPR/" rel="tag"># CRISPR</a>
              <a href="/tags/NGS/" rel="tag"># NGS</a>
              <a href="/tags/Bowtie2/" rel="tag"># Bowtie2</a>
              <a href="/tags/samtools/" rel="tag"># samtools</a>
              <a href="/tags/fastqc/" rel="tag"># fastqc</a>
              <a href="/tags/cutadapt/" rel="tag"># cutadapt</a>
              <a href="/tags/ggplot2/" rel="tag"># ggplot2</a>
              <a href="/tags/FASTA/" rel="tag"># FASTA</a>
              <a href="/tags/pipeline/" rel="tag"># pipeline</a>
              <a href="/tags/data-ana/" rel="tag"># data ana</a>
          </div>

        

          <div class="post-nav">
            <div class="post-nav-item">
                <a href="/2025/07/15/hello-world/" rel="prev" title="Hello World">
                  <i class="fa fa-angle-left"></i> Hello World
                </a>
            </div>
            <div class="post-nav-item">
                <a href="/2025/07/17/macs2/" rel="next" title="MACS2使用">
                  MACS2使用 <i class="fa fa-angle-right"></i>
                </a>
            </div>
          </div>
    </footer>
  </article>
</div>






</div>
  </main>

  <footer class="footer">
    <div class="footer-inner">

  <div class="copyright">
    &copy; 
    <span itemprop="copyrightYear">2025</span>
    <span class="with-love">
      <i class="fa fa-heart"></i>
    </span>
    <span class="author" itemprop="copyrightHolder">Meng</span>
  </div>
  <div class="powered-by">Powered by <a href="https://hexo.io/" rel="noopener" target="_blank">Hexo</a> & <a href="https://theme-next.js.org/pisces/" rel="noopener" target="_blank">NexT.Pisces</a>
  </div>

    </div>
  </footer>

  
  <div class="toggle sidebar-toggle" role="button">
    <span class="toggle-line"></span>
    <span class="toggle-line"></span>
    <span class="toggle-line"></span>
  </div>
  <div class="sidebar-dimmer"></div>
  <div class="back-to-top" role="button" aria-label="Back to top">
    <i class="fa fa-arrow-up fa-lg"></i>
    <span>0%</span>
  </div>

<noscript>
  <div class="noscript-warning">Theme NexT works best with JavaScript enabled</div>
</noscript>

</body>
</html>
