<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
<meta name="viewport" content="width=device-width">
<meta name="theme-color" content="#222"><meta name="generator" content="Hexo 7.3.0">

  <link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon-next.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/images/favicon-32x32-next.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/images/favicon-16x16-next.png">
  <link rel="mask-icon" href="/images/logo.svg" color="#222">

<link rel="stylesheet" href="/css/main.css">



<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.7.2/css/all.min.css" integrity="sha256-dABdfBfUoC8vJUBOwGVdm8L9qlMWaHTIfXt+7GnZCIo=" crossorigin="anonymous">
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/animate.css/3.1.1/animate.min.css" integrity="sha256-PR7ttpcvz8qrF57fur/yAx1qXMFJeJFiA6pSzWi0OIE=" crossorigin="anonymous">

<script class="next-config" data-name="main" type="application/json">{"hostname":"lingjunm.github.io","root":"/","images":"/images","scheme":"Pisces","darkmode":false,"version":"8.23.2","exturl":false,"sidebar":{"position":"left","width_expanded":320,"width_dual_column":240,"display":"post","padding":18,"offset":12},"hljswrap":true,"codeblock":{"theme":{"light":"default","dark":"stackoverflow-dark"},"prism":{"light":"prism","dark":"prism-dark"},"copy_button":{"enable":false,"style":null},"fold":{"enable":false,"height":500},"language":false},"bookmark":{"enable":false,"color":"#222","save":"auto"},"mediumzoom":false,"lazyload":false,"pangu":false,"comments":{"style":"tabs","active":null,"storage":true,"lazyload":false,"nav":null},"stickytabs":false,"motion":{"enable":true,"async":false,"duration":200,"transition":{"menu_item":"fadeInDown","post_block":"fadeIn","post_header":"fadeInDown","post_body":"fadeInDown","coll_header":"fadeInLeft","sidebar":"fadeInUp"}},"i18n":{"placeholder":"Searching...","empty":"We didn't find any results for the search: ${query}","hits_time":"${hits} results found in ${time} ms","hits":"${hits} results found"}}</script><script src="/js/config.js" defer></script>

    <meta name="description" content="FASTQ 格式Q&#x3D;-10*lg(p)  p为碱基出错的概率，软件计算。   p&#x3D;1%时，Q为20  0.1%对应Q30通常使用Q20作为质量标准 https:&#x2F;&#x2F;sequencing.qcfail.com&#x2F;articles&#x2F;incorrect-encoding-of-phred-scores&#x2F; In order to allow phred scores to be represent">
<meta property="og:type" content="article">
<meta property="og:title" content="FASTQ格式">
<meta property="og:url" content="https://lingjunm.github.io/2025/07/17/FASTQ%E6%A0%BC%E5%BC%8F/index.html">
<meta property="og:site_name" content="读研日记">
<meta property="og:description" content="FASTQ 格式Q&#x3D;-10*lg(p)  p为碱基出错的概率，软件计算。   p&#x3D;1%时，Q为20  0.1%对应Q30通常使用Q20作为质量标准 https:&#x2F;&#x2F;sequencing.qcfail.com&#x2F;articles&#x2F;incorrect-encoding-of-phred-scores&#x2F; In order to allow phred scores to be represent">
<meta property="og:locale" content="en_US">
<meta property="article:published_time" content="2025-07-17T03:01:42.000Z">
<meta property="article:modified_time" content="2025-08-08T08:29:53.411Z">
<meta property="article:author" content="Meng">
<meta property="article:tag" content="NGS">
<meta property="article:tag" content="FASTQ">
<meta property="article:tag" content="sanger">
<meta name="twitter:card" content="summary">


<link rel="canonical" href="https://lingjunm.github.io/2025/07/17/FASTQ%E6%A0%BC%E5%BC%8F/">


<script class="next-config" data-name="page" type="application/json">{"sidebar":"","isHome":false,"isPost":true,"lang":"en","comments":true,"permalink":"https://lingjunm.github.io/2025/07/17/FASTQ%E6%A0%BC%E5%BC%8F/","path":"2025/07/17/FASTQ格式/","title":"FASTQ格式"}</script>

<script class="next-config" data-name="calendar" type="application/json">""</script>
<title>FASTQ格式 | 读研日记</title>
  








  
  <script src="https://cdnjs.cloudflare.com/ajax/libs/animejs/3.2.1/anime.min.js" integrity="sha256-XL2inqUJaslATFnHdJOi9GfQ60on8Wx1C2H8DYiN1xY=" crossorigin="anonymous" defer></script>
<script src="/js/utils.js" defer></script><script src="/js/motion.js" defer></script><script src="/js/sidebar.js" defer></script><script src="/js/next-boot.js" defer></script>

  






  





  <noscript>
    <link rel="stylesheet" href="/css/noscript.css">
  </noscript>
</head>

<body itemscope itemtype="http://schema.org/WebPage" class="use-motion">
  <div class="headband"></div>

  <main class="main">
    <div class="column">
      <header class="header" itemscope itemtype="http://schema.org/WPHeader"><div class="site-brand-container">
  <div class="site-nav-toggle">
    <div class="toggle" aria-label="Toggle navigation bar" role="button">
        <span class="toggle-line"></span>
        <span class="toggle-line"></span>
        <span class="toggle-line"></span>
    </div>
  </div>

  <div class="site-meta">

    <a href="/" class="brand" rel="start">
      <i class="logo-line"></i>
      <p class="site-title">读研日记</p>
      <i class="logo-line"></i>
    </a>
  </div>

  <div class="site-nav-right">
    <div class="toggle popup-trigger" aria-label="Search" role="button">
    </div>
  </div>
</div>



<nav class="site-nav">
  <ul class="main-menu menu"><li class="menu-item menu-item-home"><a href="/" rel="section"><i class="fa fa-home fa-fw"></i>Home</a></li><li class="menu-item menu-item-about"><a href="/about/" rel="section"><i class="fa fa-user fa-fw"></i>About</a></li><li class="menu-item menu-item-tags"><a href="/tags/" rel="section"><i class="fa fa-tags fa-fw"></i>Tags</a></li><li class="menu-item menu-item-categories"><a href="/categories/" rel="section"><i class="fa fa-th fa-fw"></i>Categories</a></li><li class="menu-item menu-item-archives"><a href="/archives/" rel="section"><i class="fa fa-archive fa-fw"></i>Archives</a></li>
  </ul>
</nav>




</header>
        
  
  <aside class="sidebar">

    <div class="sidebar-inner sidebar-nav-active sidebar-toc-active">
      <ul class="sidebar-nav">
        <li class="sidebar-nav-toc">
          Table of Contents
        </li>
        <li class="sidebar-nav-overview">
          Overview
        </li>
      </ul>

      <div class="sidebar-panel-container">
        <!--noindex-->
        <div class="post-toc-wrap sidebar-panel">
            <div class="post-toc animated"><ol class="nav"><li class="nav-item nav-level-1"><a class="nav-link" href="#FASTQ-%E6%A0%BC%E5%BC%8F"><span class="nav-number">1.</span> <span class="nav-text">FASTQ 格式</span></a></li><li class="nav-item nav-level-1"><a class="nav-link" href="#%E6%9C%89%E5%85%B3FASTQ%E7%9A%84%E7%BC%96%E7%A0%81%E9%97%AE%E9%A2%98"><span class="nav-number">2.</span> <span class="nav-text">有关FASTQ的编码问题</span></a><ol class="nav-child"><li class="nav-item nav-level-2"><a class="nav-link" href="#1"><span class="nav-number">2.1.</span> <span class="nav-text">1</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#2"><span class="nav-number">2.2.</span> <span class="nav-text">2</span></a></li><li class="nav-item nav-level-2"><a class="nav-link" href="#3"><span class="nav-number">2.3.</span> <span class="nav-text">3</span></a></li></ol></li></ol></div>
        </div>
        <!--/noindex-->

        <div class="site-overview-wrap sidebar-panel">
          <div class="site-author animated" itemprop="author" itemscope itemtype="http://schema.org/Person">
  <p class="site-author-name" itemprop="name">Meng</p>
  <div class="site-description" itemprop="description">生物学拾遗</div>
</div>
<div class="site-state-wrap animated">
  <nav class="site-state">
      <div class="site-state-item site-state-posts">
        <a href="/archives/">
          <span class="site-state-item-count">7</span>
          <span class="site-state-item-name">posts</span>
        </a>
      </div>
      <div class="site-state-item site-state-tags">
          <a href="/tags/">
        <span class="site-state-item-count">17</span>
        <span class="site-state-item-name">tags</span></a>
      </div>
  </nav>
</div>

        </div>
      </div>
    </div>

    
  </aside>


    </div>

    <div class="main-inner post posts-expand">


  


<div class="post-block">
  
  

  <article itemscope itemtype="http://schema.org/Article" class="post-content" lang="en">
    <link itemprop="mainEntityOfPage" href="https://lingjunm.github.io/2025/07/17/FASTQ%E6%A0%BC%E5%BC%8F/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/avatar.gif">
      <meta itemprop="name" content="Meng">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="读研日记">
      <meta itemprop="description" content="生物学拾遗">
    </span>

    <span hidden itemprop="post" itemscope itemtype="http://schema.org/CreativeWork">
      <meta itemprop="name" content="FASTQ格式 | 读研日记">
      <meta itemprop="description" content="">
    </span>
      <header class="post-header">
        <h1 class="post-title" itemprop="name headline">
          FASTQ格式
        </h1>

        <div class="post-meta-container">
          <div class="post-meta">
    <span class="post-meta-item">
      <span class="post-meta-item-icon">
        <i class="far fa-calendar"></i>
      </span>
      <span class="post-meta-item-text">Posted on</span>

      <time title="Created: 2025-07-17 11:01:42" itemprop="dateCreated datePublished" datetime="2025-07-17T11:01:42+08:00">2025-07-17</time>
    </span>
    <span class="post-meta-item">
      <span class="post-meta-item-icon">
        <i class="far fa-calendar-check"></i>
      </span>
      <span class="post-meta-item-text">Edited on</span>
      <time title="Modified: 2025-08-08 16:29:53" itemprop="dateModified" datetime="2025-08-08T16:29:53+08:00">2025-08-08</time>
    </span>

  
</div>

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody"><h1 id="FASTQ-格式"><a href="#FASTQ-格式" class="headerlink" title="FASTQ 格式"></a>FASTQ 格式</h1><p><code>Q=-10*lg(p)</code>  p为碱基出错的概率，软件计算。   p&#x3D;1%时，Q为20  0.1%对应Q30<br>通常使用Q20作为质量标准</p>
<p><a target="_blank" rel="noopener" href="https://sequencing.qcfail.com/articles/incorrect-encoding-of-phred-scores/">https://sequencing.qcfail.com/articles/incorrect-encoding-of-phred-scores/</a></p>
<p>In order to allow phred scores to be represented by a single character in file formats an encoding scheme is used where ASCII values for the scores are used to represent the numerical value.  Since using raw ASCII values in the normal range of Phred scores (1 – 60 ish) would use a lot of non-printing characters an offset is applied to the phred values before doing the ASCII lookup so that more conventional characters are used.</p>
<p>Illumina used phred+64 to put the scores in the normal a-z range of ASCII, whereas the Sanger centre used phred+33 to align with the start of the normally printing ASCII characters.</p>
<p>Don’t believe the reported quality encoding on datasets.  Check the range of observed values.<br>SRR619473 in NCBIs short read archive should have phred+33 encoding (all SRA files should), but the range we found when we extracted it was clearly phred+64.  Note that this may be fixed by NCBI eventually so re-check the encoding before using this data in any training.</p>
<p>because the phred+33 and phred+64 ranges overlap it is possible to generate datasets which are compatible with both encoding schemes.  This can happen in aggressively quality trimmed phred+33 datasets where the range of values would be compatible with both schemes, with all values either being very good or very bad!</p>
<p>Some software will attempt to auto-detect the encoding used, so checking the detected encoding is a good idea.  FastQC does not allow the user to specify the quality encoding used but figures this out from the data.  The detected encoding is reported at the top of the per-base quality plot.  Phred+64 encoding is listed as illumina 1.5 (or lower).  Phred+33 is listed as Illumina 1.9&#x2F;Sanger.</p>
<p><a target="_blank" rel="noopener" href="https://hbctraining.github.io/Training-modules/planning_successful_rnaseq/lessons/QC_raw_data.html">https://hbctraining.github.io/Training-modules/planning_successful_rnaseq/lessons/QC_raw_data.html</a></p>
<p>fastqsanger, which is the scale output by Illumina since mid-2011.</p>
<h1 id="有关FASTQ的编码问题"><a href="#有关FASTQ的编码问题" class="headerlink" title="有关FASTQ的编码问题"></a>有关FASTQ的编码问题</h1><p>摘自维基百科，不同编码标准示例如下</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br></pre></td><td class="code"><pre><span class="line"> SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS.....................................................</span><br><span class="line"> ..........................XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX......................</span><br><span class="line"> ...............................IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII......................</span><br><span class="line"> .................................JJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJ.....................</span><br><span class="line"> LLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLLL....................................................</span><br><span class="line"> NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN...........................................</span><br><span class="line"> EEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEE</span><br><span class="line"> PPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPP</span><br><span class="line"> !&quot;#$%&amp;&#x27;()*+,-./0123456789:;&lt;=&gt;?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\]^_`abcdefghijklmnopqrstuvwxyz&#123;|&#125;~</span><br><span class="line"> |                         |    |        |              |               |                     |</span><br><span class="line">33                        59   64       73             88             104                   126</span><br><span class="line"> 0........................26...31.......40                                </span><br><span class="line">                          -5....0........9.............................40 </span><br><span class="line">                                0........9.............................40 </span><br><span class="line">                                   3.....9..............................41 </span><br><span class="line"> 0.2......................26...31.........41 </span><br></pre></td></tr></table></figure>

<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">S - Sanger        Phred+33,  raw reads typically (0, 40)</span><br><span class="line">X - Solexa        Solexa+64, raw reads typically (-5, 40)</span><br><span class="line">I - Illumina 1.3+ Phred+64,  raw reads typically (0, 40)</span><br><span class="line">J - Illumina 1.5+ Phred+64,  raw reads typically (3, 41)</span><br><span class="line">    with 0=unused, 1=unused, 2=Read Segment Quality Control Indicator (bold) </span><br><span class="line">    (Note: See discussion above).</span><br><span class="line">L - Illumina 1.8+ Phred+33,  raw reads typically (0, 41)</span><br><span class="line"></span><br></pre></td></tr></table></figure>

<p>Sanger format can encode a Phred quality score from 0 to 93 using ASCII 33 to 126 (although in raw read data the Phred quality score rarely exceeds 60, higher scores are possible in assemblies or read maps). Also used in SAM format.[4] Coming to the end of February 2011, Illumina’s newest version (1.8) of their pipeline CASAVA will directly produce fastq in Sanger format, according to the announcement on seqanswers.com forum.[5]<br>Solexa&#x2F;Illumina 1.0 format can encode a Solexa&#x2F;Illumina quality score from -5 to 62 using ASCII 59 to 126 (although in raw read data Solexa scores from -5 to 40 only are expected)<br>Starting with Illumina 1.3 and before Illumina 1.8, the format encoded a Phred quality score from 0 to 62 using ASCII 64 to 126 (although in raw read data Phred scores from 0 to 40 only are expected).<br>Starting in Illumina 1.5 and before Illumina 1.8, the Phred scores 0 to 2 have a slightly different meaning. The values 0 and 1 are no longer used and the value 2, encoded by ASCII 66 “B”, is used also at the end of reads as a Read Segment Quality Control Indicator.[9] The Illumina manual[10] (page 30) states the following: If a read ends with a segment of mostly low quality (Q15 or below), then all of the quality values in the segment are replaced with a value of 2 (encoded as the letter B in Illumina’s text-based encoding of quality scores)… This Q2 indicator does not predict a specific error rate, but rather indicates that a specific final portion of the read should not be used in further analyses. </p>
<p>Starting in Illumina 1.8, the quality scores have basically returned to the use of the Sanger format (Phred+33).<br>For raw reads, the range of scores will depend on the technology and the base caller used, but will typically be up to 41 for recent Illumina chemistry. Since the maximum observed quality score was previously only 40, various scripts and tools break when they encounter data with quality values larger than 40. For processed reads, scores may be even higher. For example, quality values of 45 are observed in reads from Illumina’s Long Read Sequencing Service (previously Moleculo).</p>
<p><a target="_blank" rel="noopener" href="https://www.biostars.org/p/63225/">https://www.biostars.org/p/63225/</a></p>
<p>possible ranges are 33-93 (Sanger&#x2F;Illumina1.8), 64-104 (Illumina1.3 or Illumina1.5) and 59-104 (Solexa). Similarly FastQC assumes that anything with some scores in the 33-63 range is Sanger and that the rest is Illumina1.3-1.5 (it doesn’t know about Solexa scores).</p>
<h2 id="1"><a href="#1" class="headerlink" title="1"></a>1</h2><figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">cat file.fq | awk &#x27;NR%4==0&#x27; | tr -d &#x27;\n&#x27; | hexdump -v -e&#x27;/1 &quot;%u\n&quot;&#x27; | sort -nu</span><br></pre></td></tr></table></figure>

<h2 id="2"><a href="#2" class="headerlink" title="2"></a>2</h2><p>grep Z filename # for Phred64 and make sure that the lines are not headers</p>
<h2 id="3"><a href="#3" class="headerlink" title="3"></a>3</h2><p>If the quality scores contain character 0 it is either Sanger phred+33 or Illumina 1.8+ phred+33. When they also contain the character J, it is Illumina 1.8+ phred 33, otherwise it is Sanger phred + 33.<br>When the quality scores do not contain 0, it is either Solexa +64, Illumina 1.3+ Phred+64, Illumina 1.5+ Phred+64.<br>Then it is Solexa +64 when it contains character &#x3D;<br>It is Illumina 1.3 phred + 64 when it contains A<br>It is Illumina 1.5 phred +64 when it contains no A or &#x3D;<br>Take a look at the wiki and try to understand the table</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br></pre></td><td class="code"><pre><span class="line">head -n 40 file.fastq | \</span><br><span class="line">  awk &#x27;&#123;if(NR%4==0) printf(&quot;%s&quot;,$0);&#125;&#x27; | \</span><br><span class="line">  od -A n -t u1 | \</span><br><span class="line">  awk &#x27;</span><br><span class="line">    BEGIN</span><br><span class="line">    &#123;</span><br><span class="line">      min=100;</span><br><span class="line">      max=0;</span><br><span class="line">    &#125;</span><br><span class="line"></span><br><span class="line">    &#123;</span><br><span class="line">      for(i=1;i&lt;=NF;i++) &#123;</span><br><span class="line">        if($i&gt;max) max=$i;</span><br><span class="line">        if($i&lt;min) min=$i;</span><br><span class="line">      &#125;</span><br><span class="line">    &#125;</span><br><span class="line"></span><br><span class="line">    END</span><br><span class="line">    &#123;</span><br><span class="line">      if(max&lt;=74 &amp;&amp; min&lt;59) print &quot;Phred+33&quot;;</span><br><span class="line">      else if(max&gt;73 &amp;&amp; min&gt;=64) print &quot;Phred+64&quot;;</span><br><span class="line">      else if(min&gt;=59 &amp;&amp; min&lt;64 &amp;&amp; max&gt;73) print &quot;Solexa+64&quot;;</span><br><span class="line">      else print &quot;Unknown score encoding\!&quot;;</span><br><span class="line">    &#125;</span><br><span class="line">    &#x27;</span><br></pre></td></tr></table></figure>


<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line">fqtype () &#123;</span><br><span class="line">        less $1 | head -n 999 | awk &#x27;&#123;if(NR%4==0) printf(&quot;%s&quot;,$0);&#125;&#x27; \</span><br><span class="line">        | od -A n -t u1 -v \</span><br><span class="line">        | awk &#x27;BEGIN&#123;min=100;max=0;&#125;\</span><br><span class="line">        &#123;for(i=1;i&lt;=NF;i++) &#123;if($i&gt;max) max=$i; if($i&lt;min) min=$i;&#125;&#125;END\</span><br><span class="line">        &#123;if(max&lt;=74 &amp;&amp; min&lt;59) print &quot;Phred+33&quot;; \</span><br><span class="line">        else if(max&gt;73 &amp;&amp; min&gt;=64) print &quot;Phred+64&quot;; \</span><br><span class="line">        else if(min&gt;=59 &amp;&amp; min&lt;64 &amp;&amp; max&gt;73) print &quot;Solexa+64&quot;; \</span><br><span class="line">        else print &quot;Unknown score encoding&quot;; \</span><br><span class="line">        print &quot;( &quot; min &quot;, &quot; max, &quot;)&quot;;&#125;&#x27;</span><br><span class="line">&#125;</span><br></pre></td></tr></table></figure>

<p>写在 shell 配置文件里, source 然后 fqtype <fastq_file></p>

    </div>

    
    
    

    <footer class="post-footer">
          <div class="post-tags">
              <a href="/tags/NGS/" rel="tag"># NGS</a>
              <a href="/tags/FASTQ/" rel="tag"># FASTQ</a>
              <a href="/tags/sanger/" rel="tag"># sanger</a>
          </div>

        

          <div class="post-nav">
            <div class="post-nav-item">
                <a href="/2025/07/17/%E9%93%BE%E7%89%B9%E5%BC%82%E6%80%A7/" rel="prev" title="链特异性">
                  <i class="fa fa-angle-left"></i> 链特异性
                </a>
            </div>
            <div class="post-nav-item">
                <a href="/2025/08/13/%E4%B8%8B%E8%BD%BDFASTQ%E6%95%B0%E6%8D%AE/" rel="next" title="">
                   <i class="fa fa-angle-right"></i>
                </a>
            </div>
          </div>
    </footer>
  </article>
</div>






</div>
  </main>

  <footer class="footer">
    <div class="footer-inner">

  <div class="copyright">
    &copy; 
    <span itemprop="copyrightYear">2025</span>
    <span class="with-love">
      <i class="fa fa-heart"></i>
    </span>
    <span class="author" itemprop="copyrightHolder">Meng</span>
  </div>
  <div class="powered-by">Powered by <a href="https://hexo.io/" rel="noopener" target="_blank">Hexo</a> & <a href="https://theme-next.js.org/pisces/" rel="noopener" target="_blank">NexT.Pisces</a>
  </div>

    </div>
  </footer>

  
  <div class="toggle sidebar-toggle" role="button">
    <span class="toggle-line"></span>
    <span class="toggle-line"></span>
    <span class="toggle-line"></span>
  </div>
  <div class="sidebar-dimmer"></div>
  <div class="back-to-top" role="button" aria-label="Back to top">
    <i class="fa fa-arrow-up fa-lg"></i>
    <span>0%</span>
  </div>

<noscript>
  <div class="noscript-warning">Theme NexT works best with JavaScript enabled</div>
</noscript>

</body>
</html>
